File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        lower = [0] * 26
        upper = [0] * 26
        for ch in word:
            val = ord(ch) - ord('a')
            if val < 0:
                val = ord(ch) - ord('A')
                upper[val] += 1
            else:
                lower[val] += 1
        result = 0
        for i in range(0,26):
            if lower[i] > 0 and upper[i] > 0:
                result += 1
        return result
                
        

File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.java
class Solution {
    public int numberOfSpecialChars(String word) {
        int lower[] = new int[26];
        int upper[] = new int[26];
        for(char ch : word.toCharArray()){
            int val = ch-'a';
            if(val < 0){
                val = ch-'A';
                upper[val]++;
            } else {
                lower[val]++;
            }
        }
        int result = 0;
        for(int i=0;i<26;i++){
            if(lower[i] > 0 && upper[i] > 0){
                result++;
            }
    
        }
        return result;
    }
}